openid-4-verifiable-presentations-1_0.md

```diff
@@ -460,6 +485,67 @@ To use `client_id_scheme` values `entity_id`, `did`, `verifier_attestation`, `x5
 
 Other specifications can define further values for the `client_id_scheme` parameter. It is RECOMMENDED to use collision-resistant names for such values.
 
+## Request URI Method POST {#request_uri_method_post}
+
+This request is handled by the Request URI endpoint of the Verifier.  
+
+The request MUST use the HTTP POST method with the https scheme, and the content type `application/x-www-form-urlencoded` and the accept header set to `application/oauth-authz-req+jwt`.
+
+The following parameters are defined: 
+
+`wallet_metadata`:
+: OPTIONAL. A String containing a JSON-encoded object containing metadata parameters as defined in (#as_metadata_parameters). 
+
+`wallet_nonce`:
+: OPTIONAL. A String containing a value the Verifier MUST use in the `wallet_nonce` authorization parameter when creating the signed presentation request object. For example, a base64url encoded fresh, cryptographically random number with sufficient entropy.  
+
+If the Wallet requires the Verifier to encrypt the Request Object, it SHOULD use the `jwks` or `jwks_uri` parameter within `wallet_metadata` to pass the public key for the input to the key agreement. Other mechanisms to pass the encryption key can be used as well. If the Wallet requires an encrypted Authorization Response, it SHOULD specify supported encryption algorithms using the `authorization_encryption_alg_values_supported` and `authorization_encryption_enc_values_supported` parameters. 
+
+Additionally, if the `client_id_scheme` value permits JWS Request Objects, the Wallet SHOULD list supported cryptographic algorithms for securing the Request Object through the `request_object_signing_alg_values_supported` parameter. Conversely, the Wallet MUST NOT include this parameter if the `client_id_scheme` precludes JWS Request Objects.
```

Comment:
`authorization_encryption_alg_values_supported` and `authorization_encryption_enc_values_supported` parameters are not RFC8414 parameters but are being defined in this sentence. they need to be defined either under line 497 as parameters that can be used in wallet_metadata object, or under the wallet metadata section
 
what are `JWS Request Objects`? they are not defined in rfc9101 nor in this text.
```suggestion
If the Wallet requires the Verifier to encrypt the Request Object, it SHOULD use the `jwks` or `jwks_uri` parameter within the `wallet_metadata` parameter to pass the public key for the input to the key agreement. Other mechanisms to pass the encryption key can be used as well. If the Wallet requires an encrypted Authorization Response, it SHOULD specify supported encryption algorithms using the `authorization_encryption_alg_values_supported` and `authorization_encryption_enc_values_supported` parameters. 

Additionally, if the `client_id_scheme` value permits signed Request Objects, the Wallet SHOULD list supported cryptographic algorithms for securing the Request Object through the `request_object_signing_alg_values_supported` parameter. Conversely, the Wallet MUST NOT include this parameter if the `client_id_scheme` precludes signed Request Objects.
```

Reply:
I am also a little confused with `request_object_signing_alg_values_supported` parameter because it is mentioned only once in rfc9101 (text below) and it is not defined in rfc8414. 

> the authorization server can inform the client of the algorithms that it supports in its authorization server metadata [[RFC8414](https://datatracker.ietf.org/doc/html/rfc8414)], specifically, the metadata values request_object_signing_alg_values_supported, request_object_encryption_alg_values_supported, and request_object_encryption_enc_values_supported.

Reply:
Those parameters are also used by ISO 18103-7 which is why I think it would be good to keep those. I agree it is odd that they don't show up in RFC9101 in the IANA registration section. Perhaps, we can fix this with this PR?

Reply:
Both are defined in https://openid.net/specs/openid-connect-discovery-1_0.html. I believe there was some specification that allows OIDC parameters to be used in OAuth2 too but I don't remember where this spec was. IMO, it would be good to at least add the reference to https://openid.net/specs/openid-connect-discovery-1_0.html.

Reply:
The history is a bit sorted. OIDC discovery came first. But OIDF documents can't establish an IANA registry. rfc8414 came later and established the registry https://www.iana.org/assignments/oauth-parameters/oauth-parameters.xhtml#authorization-server-metadata after which OIDC discovery retroactively made the registrations in an errata. At the bottom of https://datatracker.ietf.org/doc/html/rfc8414#section-2 AS metadata does say: 

> Additional authorization server metadata parameters MAY also be used.
>   Some are defined by other specifications, such as OpenID Connect
>   Discovery 1.0 [[OpenID.Discovery](https://datatracker.ietf.org/doc/html/rfc8414#ref-OpenID.Discovery)]. 

Then RFC9101 just refers to rfc8414 (but really I think meaning the registry it set up) and the request_object_* metadata parameters by name.

It's all confusing and not obvious. And I don't know if any of that helps this discussion. But that's the history/explanation of how that stuff came to be.

Reply:
can someone please propose text I can incorporate?